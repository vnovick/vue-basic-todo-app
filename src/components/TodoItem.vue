<template>
  <li :class="{ completed: item.completed, editing: editingTodoId === item.id }">
    <div class="view" @dblclick="setEditing(item.id)">
      <input class="toggle" type="checkbox" v-model="item.completed" />
      <label>{{item.todo}}</label>
      <button class="destroy" @click="$emit('remove-todo', item.id)"></button>
    </div>
    <input class="edit" @keyup.enter="submitUpdate(item.id)" v-model="editingTodo" />
  </li>
</template>

<script>
export default {
  data() {
    return {
      editingTodoId: null,
      editingTodo: ""
    };
  },
  props: {
    item: Object
  },
  methods: {
    setEditing(id) {
      this.editingTodoId = id;
    },
    submitUpdate(id) {
      this.$emit("submit-update", { id, editingTodo: this.editingTodo });
      this.editingTodoId = null;
      this.editingTodo = "";
    }
  }
};
</script>

<style scoped>
div {
  border: 2px solid olive;
}
</style>