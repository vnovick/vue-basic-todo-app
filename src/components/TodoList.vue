<template>
  <ul class="todo-list">
    <todo-item
      v-for="(item) in todos"
      :key="item.id"
      :item="item"
      @remove-todo="$emit('remove-todo', $event)"
      @submit-update="$emit('submit-update', $event)"
    ></todo-item>
  </ul>
</template>

<script>
import TodoItem from '../components/TodoItem'

export default {
  name: "TodoList",
  props: {
    todos: {
      type: Array,
      required: true
    }
  },
  components: {
    TodoItem
  }
};
</script>

<style lang="scss" scoped>
</style>