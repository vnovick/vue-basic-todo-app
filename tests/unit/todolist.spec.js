import TodoList from '@/components/TodoList'
import TodoItem from '@/components/TodoItem'
import { mount } from '@vue/test-utils';


describe('Todo List spec', () => {
    it('Empty component renders correctly', () => {
        const todos = []

        const wrapper = mount(TodoList, {
            propsData: { todos },
        })
        expect(wrapper.find('.todo-list')).toBeDefined()
        expect(wrapper).toMatchSnapshot()
    })
    //Snapshots
    it('Component with one todo renders correctly', () => {
        const todos = [{
            id: 1,
            todo: 'First todo'
        }]

        const dummyRemove = jest.fn()
        const dispatch = jest.fn()

        const wrapper = mount(TodoList, {
            propsData: { todos },
            methods: {
                dummyRemove
            },
            $store: {
                dispatch
            }
        })

        wrapper.find(TodoItem).vm.$emit('dummy-remove')
        expect(dummyRemove).toBeCalled()
    })
})