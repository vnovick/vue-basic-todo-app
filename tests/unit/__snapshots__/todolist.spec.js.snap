// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo List spec Component with one todo renders correctly 1`] = `<span><ul class="todo-list"><todo-item-stub item="[object Object]"></todo-item-stub></ul></span>`;

exports[`Todo List spec Empty component renders correctly 1`] = `<span><ul class="todo-list"></ul></span>`;
